comp110-worksheet-7/DirectoryUtils.cs

```diff
@@ -24,37 +27,123 @@ public static bool IsDirectory(string path)
+			long totalSize = 0;
+			// Gets all files in the directory
+			string[] filePaths = Directory.GetFiles(directory, "*", SearchOption.AllDirectories);
+			// Iterates through each item in the array
+			for (int i = 0; i < filePaths.Length; i++)
+			{
+				// Adds the size of each item to the total size
+				totalSize += new FileInfo(filePaths[i]).Length;
+			}
+			// Returns the final total for totalSize
+			return totalSize;
 		}
 
 		// Return the number of files (not counting directories) below the given directory
 		public static int CountFiles(string directory)
 		{
-			throw new NotImplementedException();
+			// Get all files in the directory
+			string[] filePaths = Directory.GetFiles(directory, "*", SearchOption.AllDirectories);
+			// Returns the size of the array
+			return filePaths.Length;
 		}
 
 		// Return the nesting depth of the given directory. A directory containing only files (no subdirectories) has a depth of 0.
-		public static int GetDepth(string directory)
+		public static int GetDepth(string directory, int currentDepth = 0)
 		{
-			throw new NotImplementedException();
+			// Defines a new list that is unique to this run of the function
+			List<int> depthValues = new List<int>();
+
+			// Each directory is collected for use in iteration of this current run of the function
+			string[] directoryDepth = Directory.GetDirectories(directory);
+
+			// If the directory has more sub directories, then the current depth is incremented by 1
+			if(directoryDepth.Length > 0)
+			{
+				currentDepth++;
+			}
+			// Loops through each of the subdirectories of this current run of the function
+			for(int i = 0; i < directoryDepth.Length; i++)
+			{
+				// Adds the currentDepth values of each subdirectory to a list
+				depthValues.Add(GetDepth(directoryDepth[i], currentDepth));
+			}
+			// For each of the items in the depthValues list...
+			for(int i = 0; i < depthValues.Count; i++)
+			{
+				// If the value is more than the currentDepth for the original directory path
+				if (depthValues[i] > currentDepth)
+				{
+					// It becomes the new currentDepth value
+					currentDepth = depthValues[i];
+				}
+			}
+			
+			return currentDepth;
 		}
 
 		// Get the path and size (in bytes) of the smallest file below the given directory
 		public static Tuple<string, long> GetSmallestFile(string directory)
 		{
-			throw new NotImplementedException();
+			// Get all filePaths in the directory
+			string[] filePaths = Directory.GetFiles(directory, "*", SearchOption.AllDirectories);
+			string smallestFile = "";
+
+			// Iterates through every item in filePaths
+			for(int i = 0; i < filePaths.Length; i++)
+			{
+				// The first item is assigned as the smallestFile
+				if (smallestFile == "")
+				{
+					smallestFile = filePaths[i];
+				}
+				// Each next file is checked against the size of the current smallestFile
+				else if (new FileInfo(filePaths[i]).Length < new FileInfo(smallestFile).Length)
```

Comment:
You are recomputing `new FileInfo(smallestFile).Length` a lot here -- would be more efficient to store and reuse the value